src/main/scala/io/shiftleft/fuzzyc2cpg/passes/CfgCreationPass.scala

```diff
@@ -0,0 +1,383 @@
+
+  override def partIterator: Iterator[nodes.Method] = cpg.method.iterator
+
+  override def runOnPart(method: nodes.Method): Iterator[DiffGraph] =
+    new CfgCreatorForMethod(method).run()
+
+}
+
+class CfgCreatorForMethod(entryNode: nodes.Method) {
+
+  private implicit class FringeWrapper(fringe: List[FringeElement]) {
+    def setCfgEdgeType(cfgEdgeType: CfgEdgeType): List[FringeElement] = {
+      fringe.map {
+        case FringeElement(node, _) =>
+          FringeElement(node, cfgEdgeType)
+      }
+    }
+    def add(node: nodes.CfgNode, cfgEdgeType: CfgEdgeType): List[FringeElement] =
+      FringeElement(node, cfgEdgeType) :: fringe
+
+    def add(ns: List[nodes.CfgNode], cfgEdgeType: CfgEdgeType): List[FringeElement] =
+      ns.map(node => FringeElement(node, cfgEdgeType)) ++ fringe
+
+    def add(otherFringe: List[FringeElement]): List[FringeElement] =
+      otherFringe ++ fringe
+  }
+
+  private val logger = LoggerFactory.getLogger(getClass)
+  val diffGraph: DiffGraph.Builder = DiffGraph.newBuilder
+
+  private var fringe = List[FringeElement]().add(entryNode, AlwaysEdge)
+  private var markerStack = List[Option[nodes.CfgNode]]()
+  private case class FringeElement(node: nodes.CfgNode, cfgEdgeType: CfgEdgeType)
+  private var labeledNodes = Map[String, nodes.CfgNode]()
+  private var pendingGotoLabels = List[String]()
+  private var pendingCaseLabels = List[String]()
+  private var returns = List[nodes.CfgNode]()
+  private val breakStack = new LayeredStack[nodes.CfgNode]()
+  private val continueStack = new LayeredStack[nodes.CfgNode]()
+  private val caseStack = new LayeredStack[(nodes.CfgNode, Boolean)]()
+  private var gotos = List[(nodes.CfgNode, String)]()
+
+  def run(): Iterator[DiffGraph] = {
+    postOrderLeftToRightExpand(entryNode)
+    connectGotosAndLabels()
+    connectReturnsToExit()
+    Iterator(diffGraph.build)
+  }
+
+  private def postOrderLeftToRightExpand(node: nodes.AstNode): Unit = {
+    node match {
+      case n: nodes.ControlStructure =>
+        handleControlStructure(n)
+      case n: nodes.JumpTarget =>
+        handleJumpTarget(n)
+      case call: nodes.Call if call.name == Operators.conditional =>
+        handleConditionalExpression(call)
+      case call: nodes.Call if call.name == Operators.logicalAnd =>
+        handleAndExpression(call)
+      case call: nodes.Call if call.name == Operators.logicalOr =>
+        handleOrExpression(call)
+      case call: nodes.Call =>
+        expandChildren(call)
+        extendCfg(call)
+      case identifier: nodes.Identifier =>
+        extendCfg(identifier)
+      case literal: nodes.Literal =>
+        extendCfg(literal)
+      case actualRet: nodes.Return =>
+        expandChildren(actualRet)
+        extendCfg(actualRet)
+        fringe = Nil
+        returns = actualRet :: returns
+      case formalRet: nodes.MethodReturn =>
+        extendCfg(formalRet)
+      case n: nodes.AstNode =>
+        expandChildren(n)
```

Comment:
For the sake of uniformity i would suggest creating a "handle" function for all kind of nodes.

Reply:
:+1: